stripe/src/main/java/com/stripe/android/net/StripeApiHandler.java

```diff
@@ -419,11 +477,15 @@ static String createQuery(Map<String, Object> params)
         JSONObject headerMappingObject = new JSONObject(propertyMap);
         headers.put("X-Stripe-Client-User-Agent", headerMappingObject.toString());
 
-        if (apiVersion != null) {
-            headers.put("Stripe-Version", apiVersion);
+        if (options != null && options.getApiVersion() != null) {
+            headers.put("Stripe-Version", options.getApiVersion());
         }
 
-        if (options.getIdempotencyKey() != null) {
+        if (options != null && options.getGuid() != null) {
+            headers.put("guid", options.getGuid());
```

Comment:
This is actually used for the API and for the m.stripe request. I get all my headers in the same place, and the "guid" value is only non-null / non-empty for the m.stripe ones.